modin/core/execution/dask/common/engine_wrapper.py

```diff
@@ -133,3 +134,29 @@ def put(cls, data, **kwargs):
             data = UserDict(data)
         client = default_client()
         return client.scatter(data, **kwargs)
+
+    @classmethod
+    def wait(cls, obj_ids, num_returns=None):
+        """
+        Wait on the objects without materializing them (blocking operation).
+
+        Parameters
+        ----------
+        obj_ids : list, scalar
+        num_returns : int, optional
+        """
+        if not isinstance(obj_ids, list):
+            obj_ids = [obj_ids]
+        if num_returns is None:
+            num_returns = len(obj_ids)
+        if num_returns == len(obj_ids):
+            wait(obj_ids, return_when="ALL_COMPLETED")
+        else:
+            # Dask doesn't natively support `num_returns` as int.
+            # `wait` function doesn't always return only one finished future,
+            # so a simple loop is not enough here
+            done, not_done = wait(obj_ids, return_when="FIRST_COMPLETED")
+            while len(done) < num_returns and (i := 0 < num_returns):
```

Comment:
Counter just in case, so as not to go into an infinite loop under any circumstances

Reply:
I don't see the second while loop as in unidist. There might still be a case when

Reply:
Which circumstances do you mean? This counter feels redundant.

Reply:
> Which circumstances do you mean?

Any bugs in `wait` function that could lead to an infinite loop here.